test/cppsuite/tests/cursor_bound_01.cpp

```diff
@@ -290,6 +290,48 @@ class cursor_bound_01 : public test {
         return std::make_pair(lower_bound, upper_bound);
     }
 
+    /*
+     * Validate the bound search call. If the key is within range, the range cursor should return
+     * back the search key.
+     */
+    void
+    validate_bound_search(int range_ret, scoped_cursor &range_cursor, scoped_cursor &normal_cursor,
+      const std::string &search_key, const bound &lower_bound, const bound &upper_bound)
+    {
+
+        auto lower_key = lower_bound.get_key();
+        auto upper_key = upper_bound.get_key();
+        auto lower_inclusive = lower_bound.get_inclusive();
+        auto upper_inclusive = upper_bound.get_inclusive();
+
+        const char *key;
+        testutil_check(range_cursor->get_key(range_cursor.get(), &key));
+        logger::log_msg(LOG_TRACE,
+          "bounded search found key: " + std::string(key) + " with lower bound: " + lower_key +
+            " upper bound: " + upper_key);
+
+        /*
+         * Assert that if range cursor returned with a key, the search key has to be within range.
+         */
+        auto above_lower_key = lower_key.empty() ||
+          custom_lexicographical_compare(lower_key, search_key, lower_inclusive);
+        auto below_upper_key = upper_key.empty() ||
+          custom_lexicographical_compare(search_key, search_key, upper_inclusive);
```

Comment:
Seems that we have `search_key` twice here, is it expected?

Reply:
Nope. Fixed that!